src/compiler/scala/tools/nsc/typechecker/RefChecks.scala

```diff
@@ -358,14 +358,15 @@ abstract class RefChecks extends Transform {
 
         def overrideError(msg: String) {
           if (noErrorType)
-            emitOverrideError(overrideErrorMsg(msg))
+            emitOverrideError(overrideErrorMsg(infoString(member) + " " + msg))
         }
 
         def overrideTypeError() {
           if (noErrorType) {
             emitOverrideError(
               if (member.isModule && other.isModule) objectOverrideErrorMsg
-              else overrideErrorMsg("has incompatible type")
+              else if (!other.isDeferred) overrideErrorMsg(s"overriding concrete type alias ${infoString(member)} is not allowed except when equivalent")
+              else overrideErrorMsg(s"${infoString(member)} has incompatible type")
```

Comment:
My limited ability to compose strings in my head suggests we could use two test cases in the test, for each branch.

At least I can do code coverage in my head.